src/gobimport/reader.py

```diff
@@ -94,23 +94,29 @@ def _query(self, query):
         else:
             yield from query
 
-    def read(self):  # noqa: C901
+    def read(self, mode=FULL_IMPORT):  # noqa: C901
         """Read the data from the data source
 
         :return: iterable dataset
         """
         assert self._connection is not None, "No connection, connect should succeed before read"
 
+        # The source query is the query (only db-like connections have one)
+        source_query = self.source.get("query", [])
+        if mode != FULL_IMPORT:
+            # Optionally populated with the mode, eg partial, random, ...
+            source_query += self.source[mode]
```

Comment:
If the wrong mode is passed a KeyError will occor. Raise GOBException in that case with a 'nice' error message?

Reply:
Changed this one